esphome/components/mqtt/mqtt_climate.cpp

```diff
@@ -97,6 +97,8 @@ void MQTTClimateComponent::send_discovery(JsonObject &root, mqtt::SendDiscoveryC
       fan_modes.add("focus");
     if (traits.supports_fan_mode(CLIMATE_FAN_DIFFUSE))
       fan_modes.add("diffuse");
+    for (auto fan_mode : traits.get_supported_custom_fan_modes())
```

Comment:
```suggestion
    for (const auto &fan_mode : traits.get_supported_custom_fan_modes())
```

See also the clang-tidy error.